src/lib/language-utils.ts

```
@@ -0,0 +1,88 @@
+  ttl: 1000 * 60 * 60,
+});
+
+export const normalize = (text: string): string => {
+  if (NORMALIZATION_CACHE.has(text)) {
+    return NORMALIZATION_CACHE.get(text)!;
+  }
+
+  // Normalize Unicode and remove zero-width/invisible chars
+  let normalized = text
+    .normalize("NFC")
+    .replace(/[\u200B-\u200D\uFEFF\u00AD]/g, "")
```

Biome (1.9.4)

[error] A character class cannot match a joined character sequence.

A zero width joiner composes several emojis into a new one. Replace the character class with an alternation.

(lint/suspicious/noMisleadingCharacterClass)